pkg/restapi/issuer/operation/operations.go

```diff
@@ -1139,22 +1148,77 @@ func (o *Operation) storeUserInvitationMapping(mapping *UserInvitationMapping) e
 }
 
 // TODO fulfillment shouldn't be saved, need re-design [Issue#560]
-func (o *Operation) saveCredentialFulfillment(thID string, fulfillment *verifiable.Presentation) error {
-	fbytes, err := fulfillment.MarshalJSON()
+func (o *Operation) saveCredentialAttachments(thID string, attachments *credentialAttachments) error {
+	manifestBytes, err := json.Marshal(attachments.Manifest)
 	if err != nil {
-		return fmt.Errorf("failed to marshal credential fulfillment: %w", err)
+		return fmt.Errorf("failed to marshal credential manifest: %w", err)
 	}
 
-	err = o.txnStore.Put(thID, fbytes)
+	fulfilmentBytes, err := json.Marshal(attachments.Fulfilment)
+	if err != nil {
+		return fmt.Errorf("failed to marshal credential fulfilment: %w", err)
+	}
+
+	err = o.txnStore.Batch([]storage.Operation{
+		{
+			Key:   manifestPrefix + thID,
+			Value: manifestBytes,
+		},
+		{
+			Key:   fulfilmentPrefix + thID,
+			Value: fulfilmentBytes,
+		},
+	})
 	if err != nil {
-		return fmt.Errorf("failed to save credential fulfillment in store: %w", err)
+		return fmt.Errorf("failed to save credential attachments in store: %w", err)
 	}
 
 	return nil
 }
 
+func (o *Operation) readAndValidateCredentialApplication(msg service.DIDCommAction) (*cm.CredentialApplication, error) {
+	applicationAttachment, err := getAttachments(msg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to correlate WACI interaction, missing thread ID: %w", err)
+	}
+
+	if len(applicationAttachment) != 1 {
+		return nil, errors.New("waci request credential: expected only one attachment for request credential")
+	}
+
+	credentialApplicationBytes, err := getCredentialApplicationFromAttachment(&applicationAttachment[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to get credential application from attachments: %w", err)
+	}
+
+	if credentialApplicationBytes != nil {
+		credManifestBytes, err := o.txnStore.Get(manifestPrefix + msg.Message.ID())
+		if err != nil {
+			return nil, fmt.Errorf("failed to get persisted credential manifest attachement: %w", err)
+		}
+
+		credManifest := credentialAttachments{}.Manifest
```

Comment:
No need to create an entire `credentialAttachments` object just to get a `CredentialManifest` object.  Instead, just create a `*cm.CredentialManifest` directly by doing

`credManifest := &cm.CredentialManifest{}`

OR

`var credManifest *cm.CredentialManifest{}`